### PR TITLE
Feat/market details

### PR DESCRIPTION
**Pull request by djaciel**

### CODE REVIEW

File: packages/overlay-sdk/src/services/marketsDetails.ts [hunk 1]
```diff
@@ -34,3 +35,38 @@ export const getMarketsDetailsByChainId = async (chainId: CHAINS) => {
...
+    const marketDetail = {
+      marketName: marketDetailsData.marketName,
+      disabled: marketDetailsData.chains[0].disabled,
+      logo: marketDetailsData.logo,
+      currency: marketDetailsData.currency,
+      descriptionText: marketDetailsData.descriptionText,
+      fullLogo: marketDetailsData.fullLogo,
+      oracleLogo: marketDetailsData.oracleLogo,
+      indexesConstruction: marketDetailsData.indexesConstruction || [],
+      chains: marketDetailsData.chains.map((chain) => {
```
Comment (Aliona-D) [inline packages/overlay-sdk/src/services/marketsDetails.ts:55]: Here should add filter and leave only data with current chainId

File: packages/overlay-sdk/src/marketsList/markets.ts [hunk 2]
```diff
@@ -6,16 +6,65 @@ import { MARKET_LOGO, PRICE_CURRENCY_FROM_QUOTE, V1_PERIPHERY_ADDRESS, ORACLE_LO
...
+
+  public async getMarketDetails(marketId: string) {
+    const chainId = this.core.chainId
+    invariant(chainId in CHAINS, "Unsupported chainId");
+
+    const marketDetails = await getMarketDetailsById(marketId, chainId)
+    invariant(marketDetails, "Market not found");
+
+    const contract = marketDetails.chains.filter((chain) => chain.chainId === chainId)[0].deploymentAddress
+    const result = await this.sdk.state.getMarketState(V1_PERIPHERY_ADDRESS[chainId], contract as Address)
```
Comment (Aliona-D) [inline packages/overlay-sdk/src/marketsList/markets.ts:27]: result and all further transformations shold be removed from here

File: packages/overlay-sdk/src/marketsList/markets.ts [hunk 3]
```diff
@@ -6,16 +6,65 @@ import { MARKET_LOGO, PRICE_CURRENCY_FROM_QUOTE, V1_PERIPHERY_ADDRESS, ORACLE_LO
...
+      parsedOiLong = decimals && formatBigNumber(result.oiLong, decimals, 18)
+      parsedOiShort = decimals && formatBigNumber(result.oiShort, decimals, 18)
+      parsedCapOi = decimals && formatBigNumber(result.capOi, decimals, 18)
+      parsedDailyFundingRate = decimals && formatFundingRateToDaily(result.fundingRate, 18, 2)
+      parsedAnnualFundingRate = decimals && formatFundingRateToAnnual(result.fundingRate, 18, 2)
+
+      return {
+        ...marketDetails,
+        ...result,
+        contract,
```
Comment (Aliona-D) [inline packages/overlay-sdk/src/marketsList/markets.ts:53]: return only marketDetails and contract (should be renamed to marketAddress)

File: packages/overlay-sdk/src/trade/trade.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    const result = await this.sdk.state.getMarketState(V1_PERIPHERY_ADDRESS[chainId], marketAddress as Address)
+    invariant(result, "Market state not found");
+
+    let parsedOiLong: string | number | undefined = undefined
+    let parsedOiShort: string | number | undefined = undefined
+    let parsedCapOi: string | number | undefined = undefined
+
+    const decimals = 18
+
+    parsedOiLong = decimals && formatBigNumber(result.oiLong, decimals, 18)
```
Comment (Aliona-D) [inline packages/overlay-sdk/src/trade/trade.ts:45]: here decimals is also constant

File: packages/overlay-sdk/src/trade/trade.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      estimatedPrice = await this.sdk.state.getBid(V1_PERIPHERY_ADDRESS[chainId], marketAddress as Address, fractionOfCapOi)
+    }
+
+    let parsedEstimatedPrice: string | number | undefined = undefined
+    parsedEstimatedPrice = formatBigNumber(estimatedPrice, 18, 5)
+
+    return parsedEstimatedPrice
+  }
+
+  public async getBidAndAsk(marketId: Address, collateral?: bigint, leverage?: bigint, isLong?: boolean) {
```
[the remark this answers is not in the document]
Comment (djaciel) [reply packages/overlay-sdk/src/trade/trade.ts:86]: typo; should be marketAddress as the other methods

File: packages/overlay-sdk/src/trade/trade.ts [hunk 6]
```diff
@@ -69,7 +72,34 @@ export class OverlaySDKTrade extends OverlaySDKModule {
...
     parsedEstimatedPrice = formatBigNumber(estimatedPrice, 18, 5)
 
-    return parsedEstimatedPrice
+    return {
+      rawNumber: estimatedPrice,
+      parsedNumber: parsedEstimatedPrice
+    }
+  }
+
+  public async getPriceInfo(marketId: string, collateral: bigint, leverage: bigint, slippage: number, isLong: boolean) {
```
Comment (Aliona-D) [inline packages/overlay-sdk/src/trade/trade.ts:81]: collateral, leverage and slippage are strings

File: packages/overlay-sdk/src/trade/trade.ts [hunk 7]
```diff
@@ -148,4 +148,15 @@ export class OverlaySDKTrade extends OverlaySDKModule {
 
     return tradingFeeRate * 100
   }
+
+  public async getLiquidationPriceEstimate(marketId: string, collateral: bigint, leverage: bigint, isLong: boolean) {
```
Comment (Aliona-D) [inline packages/overlay-sdk/src/trade/trade.ts:152]: collateral, leverage are strings

File: packages/overlay-sdk/src/trade/trade.ts [hunk 8]
```diff
@@ -159,4 +159,15 @@ export class OverlaySDKTrade extends OverlaySDKModule {
 
     return formatBigNumber(liquidationPrice, 18, 5)
   }
+
+  public async getOiEstimate(marketId: string, collateral: bigint, leverage: bigint, isLong: boolean) {
```
Comment (Aliona-D) [inline packages/overlay-sdk/src/trade/trade.ts:163]: collateral, leverage - strings

File: apps/overlay/src/pages/Markets/marketsTable.js [hunk 9]
```diff
@@ -17,6 +17,7 @@ import {
 import useSDK from "../../hooks/useSDK";
 import { useMultichainContext } from "../../state/multichain/useMultichainContext";
 import { useAccount } from "../../hooks/useAccount";
+import { toWei } from "overlay-sdk/dist/common/utils/formatWei";
```
Comment (Aliona-D) [inline apps/overlay/src/pages/Markets/marketsTable.js:20]: In application it should be imported toWei function the same way? Maybe it is possible to export it in constructor(sdk)?

**Comment by Aliona-D** [comment]
In src/sdk.ts export of sdk(L32-43) should be removed. Also in constructor there should be only modules. I think midPrice is not a module, it is a method, thus it should be removed from there.

**Pull request merged**